### ISSUE TITLE
Mapping owned type issue in Postgresql

### ISSUE THREAD

**Issue by sbabaei** [efcore/EFCore.NamingConventions#50]
Another Bug! I have an Owned Type that I want to map to the same table as the owner.
This a code snippet of my configuration:

        builder.OwnsOne<ImportanceDegree>(publication => publication.ImportanceDegree, a =>
        {
            a.Property(x => x.SeverityId).HasColumnName("SeverityId");
        });
When I adding migration I get this error:

The keys {'Id'} on 'Publication' and {'PublicationId'} on 'ImportanceDegree' are both mapped to 'Publications.Publication.pk_publication' but with different columns ({'Id'} and {'id'}).

I do not have any relation or a foreign key. I just want to map a property in the same table as the owner.

**Comment by roji** [comment]
@sbabaei can you please submit a runnable code sample, including a minimal model? It's difficult to know exactly what you're doing from the fragment above.

**Comment by hey-red** [comment]
Same issue after upgrade from 5.0.0-rc1 to 5.0.0.

```C#
// Entity
public class Attachment 
{
    public int Id { get; set; }
    public Dimensions? Dimensions { get; private set; }
}

// Owned
public record Dimensions(int Width, int Height);

// Entity configuration
public class AttachmentEntityTypeConfiguration : IEntityTypeConfiguration<Attachment>
{
    public void Configure(EntityTypeBuilder<Attachment> builder)
    {
        builder.OwnsOne(m => m.Dimensions);
    }
}

// AppDbContext
protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder) => optionsBuilder.UseSnakeCaseNamingConvention();
```

`The keys {'AttachmentId'} on 'Dimensions' and {'Id'} on 'Attachment' are both mapped to 'attachments.pk_attachments' but with different columns ({'attachment_id'} and {'id'}).`

**Comment by AdamQuintana** [comment]
I am seeing a similar issue with my owned entities.

```
//entity
public class Asr : Base, IEntity
{
	public long Id { get; set; }
	
	...
	...
	
	public AsrEntity Entity { get; set; }
}

//owned
public class AsrEntity : ValueObject
{
	public string Name { get; private set; }
	
	...
	...
}

//configuration
public class AsrsConfiguration : IEntityTypeConfiguration<Asr>
{
	public void Configure(EntityTypeBuilder<Asr> entity)
	{
		entity.OwnsOne(r => r.Entity, n =>
		{
			n.Property(e => e.Name)
				.HasColumnName("entity_owner_name");
			
			...
			...
		});
	}
}
```
Migration works in 5.0.0-rc1, but get the following error with 5.0.0:

The keys {'AsrId'} on 'AsrEntity' and {'Id'} on 'Asr' are both mapped to 'asrs.pk_asrs' but with different columns ({'asr_id'} and {'id'}).

**Comment by jdcrutchley** [comment]
It's kind of ugly, but you can work around this by explicitly mapping the owned type's foreign key shadow property to the owning type's id column - something like:

`
public void Configure(EntityTypeBuilder<OwnerType> builder)
{
  builder.OwnsOne(ot => ot.OwnedTypeProperty =>
  {
    p => p.Property("OwnerTypeId").HasColumnName("id");
  }
}
`

**Comment by awajan** [comment]
Having same after upgrading library from 5-rc1 to 5

**Comment by roji** [comment]
Thanks to everyone reporting, I can reproduce this and will fix it for a 5.0.1 release which should happen very soon.
